assets/js/vue/img.js

```diff
@@ -17,60 +24,106 @@ export default {
+  methods: {
+    mergeSettings() {
+      const defaultSettings = {
+        cloud_name: '',
+        developer_mode: false,
+        verbose: false,
+      };
+      return Object.assign(defaultSettings, this.cloudinary);
     },
-    height() {
-      // TODO implement height calculation
+    setPreset() {
+      const settings = this.mergeSettings();
+      const presetExists = settings['presets'] && settings['presets']['default'];
+      return presetExists ? Object.assign(this.defaultPresets, settings['presets']['default']) : this.defaultPresets;
     },
-    urlParams() {
-      return this.getUrlParams();
+    getCloudinaryLink(url) {
+      return basePath + url;
     },
-  },
-  methods: {
-    isGif() {
-      const extension = this.img.split('.')[1];
+    getMeta(url) {
+      return new Promise((resolve, reject) => {
+        let img = new Image();
+        img.onload = () => resolve(img);
+        img.onerror = (err) => reject(err);
+        img.src = this.getCloudinaryLink(url);
+      });
+    },
+    async initialize() {
+      if (this.cloudinary && !this.isGif()) {
+        let img = await this.getMeta(this.img);
+        const height = img?.height;
+        const width = img?.width;
 
-      return extension.toLowerCase() === 'gif';
+        const preset = this.setPreset();
+        const transformationsString = this.getTransformationString(preset);
+
+        if (height && width) {
+          this.dimensions = [height, width];
+          this.buildSrcSet(preset, transformationsString);
+        } else {
+          this.dimensions = [preset.fallback_max_width, preset.fallback_max_width];
+          this.fallback = `${basePath}${transformationsString},w_${preset.fallback_max_width}/${this.img} ${width}w`;
+        }
+      }
     },
-    getPreset() {
-      const selectedPreset = Presets[this.preset];
+    getTransformationString(preset) {
+      const transformations = [];
+      for (const [key, value] of Object.entries(this.transformationOptions)) {
+        if (preset[key]) {
+          transformations.push(`${value}_${preset[key]}`);
+        }
+      }
+      const noTransformations = transformations.length === 0;
 
-      return selectedPreset ? selectedPreset : Presets['default'];
+      return !noTransformations ? transformations.join(',') : '';
     },
-    getPresetWidth(value) {
-      const width = value ? value : this.getPreset()?.fallback_max_width;
+    buildSrcSet(preset, transformationsString) {
+      const srcsetArray = [];
+      const steps = preset['steps'];
+      const minWidth = preset['min_width'];
+      const maxWidth = preset['max_width'];
+      const stepWidth = (maxWidth - minWidth) / (steps - 1);
+      const [_, naturalWidth] = this.dimensions;
 
-      if (!width) return '';
+      if (naturalWidth < minWidth) {
+        const srcsetString = '';
+        srcsetArray.push(srcsetString);
+      } else {
+        let missedSizes = [];
+        for (let factor = 1; factor < steps; factor++) {
```

Comment:
Leerzeilen zur Lesbarkeit würden hier beim Verstehen helfen.

Was hat es mit den missedSizes auf sich? Verstehe die Logik nicht ganz

Reply:
Im Original Code sind die missedSizes. Die Größen (bzw. widths), die größer sind als unser Original Bild, aber sich noch in der von uns definierten Stepweite befindet. Diese werden dann nochmal zusätzlich ins srcset übertragen